src/openrct2/localisation/Localisation.cpp

```diff
@@ -1138,6 +1138,13 @@ static void format_string_code(uint32_t format_code, char** dest, size_t* size,
 
             format_length(dest, size, static_cast<uint16_t>(value));
             break;
+        case FORMAT_LENGTH_SIGNED:
+            // Pop argument
+            std::memcpy(&value, *args, sizeof(int16_t));
+            *args += 2;
+
+            format_length(dest, size, static_cast<int16_t>(value));
+            break;
```

Comment:
To be honest, I am not a fan of introducing a new format code for this. Can't length always be signed instead? If really necessary, we can increase it to a 32-bits value.

Reply:
I agree, as I am planning on refactoring out the arg buffer which eliminates strict sizes altogether. This will mean that in the language strings, we will be left with `{COMMA}`, `{INT}`, `{LENGTH}`.

Reply:
Thank you all for the feedback. I have ben having fun looking into this project.

@AaronVanGeffen @duncanspumpkin 

I see where you are coming from and agree. I think to extend the length format to 32-bits I'd need to find all the places which ultimately use the length formatter and ensure they are sending in a pointer to a 32-bit value. Such as [here](https://github.com/OpenRCT2/OpenRCT2/blob/0bf851cfb290636969a95b9d84528962c536df82/src/openrct2-ui/windows/Ride.cpp#L6006-L6011). Is that your understanding as well?

@IntelOrca 

That sounds much more usable. Was that something you were planning to tackle relatively soon or just something that is planned for future work? If it is already in the works, perhaps it'd be best for me to just let your refactor resolve this issue.

Reply:
I am currently developing it. I wouldn't let that put you off from finding every occurrence that length is used and replacing the int16 with an int32. I doubt the are too many usages of LENGTH.